test/Microsoft.AspNetCore.JsonPatch.Test/JsonPatchDocumentJsonPropertyAttributeTest.cs

```diff
@@ -80,142 +89,157 @@ public void Add_WithExpressionOnStringProperty_FallsbackToPropertyName_WhenJsonP
         public void Add_WithExpression_RespectsJsonPropertyName_WhenApplyingToSameTypedClassWithMatchingJsonPropertyName()
         {
+            // Arrange
             var patchDocToSerialize = new JsonPatchDocument<JsonPropertyObject>();
             patchDocToSerialize.Add(p => p.Name, "John");
 
-            // the patchdoc will deserialize to "anothername".  As JsonPropertyDTO has
-            // a JsonProperty signifying that "Name" should be deseriallized from "AnotherName",
-            // we should be able to apply the patchDoc.
-
-            var doc = new JsonPropertyObject()
+            var targetObject = new JsonPropertyObject()
             {
                 Name = "InitialValue"
             };
 
             var serialized = JsonConvert.SerializeObject(patchDocToSerialize);
+
+            // Act
             var deserialized =
                 JsonConvert.DeserializeObject<JsonPatchDocument<JsonPropertyObject>>
                 (serialized);
 
-            deserialized.ApplyTo(doc);
+            deserialized.ApplyTo(targetObject);
 
-            Assert.Equal("John", doc.Name);
+            // Assert
+            Assert.Equal("John", targetObject.Name);
         }
 
         [Fact]
         public void Add_OnApplyFromJson_RespectsJsonPropertyNameOnJsonDocument()
         {
-            var doc = new JsonPropertyObject()
+            // Arrange
+            var targetObject = new JsonPropertyObject()
             {
                 Name = "InitialValue"
             };
 
             // serialization should serialize to "AnotherName"
             var serialized = "[{\"value\":\"Kevin\",\"path\":\"/AnotherName\",\"op\":\"add\"}]";
+
+            // Act
             var deserialized =
                 JsonConvert.DeserializeObject<JsonPatchDocument<JsonPropertyObject>>(serialized);
+            deserialized.ApplyTo(targetObject);
 
-            deserialized.ApplyTo(doc);
-
-            Assert.Equal("Kevin", doc.Name);
+            // Assert
+            Assert.Equal("Kevin", targetObject.Name);
         }
 
         [Fact]
         public void Remove_OnApplyFromJson_RespectsJsonPropertyNameOnJsonDocument()
         {
-            var doc = new JsonPropertyObject()
+            // Arrange
+            var targetObject = new JsonPropertyObject()
             {
                 Name = "InitialValue"
             };
 
             // serialization should serialize to "AnotherName"
             var serialized = "[{\"path\":\"/AnotherName\",\"op\":\"remove\"}]";
+
+            // Act
             var deserialized =
                 JsonConvert.DeserializeObject<JsonPatchDocument<JsonPropertyObject>>(serialized);
+            deserialized.ApplyTo(targetObject);
 
-            deserialized.ApplyTo(doc);
-
-            Assert.Null(doc.Name);
+            // Assert
+            Assert.Null(targetObject.Name);
         }
 
         [Fact]
         public void Add_OnApplyFromJson_RespectsInheritedJsonPropertyNameOnJsonDocument()
```

Comment:
>Add_OnApplyFromJson_RespectsInheritedJsonPropertyNameOnJsonDocument [](start = 20, length = 67)

I don't think we need as many of these tests as there are. This is just re-testing how JsonContract behaves. We probably need 1-2 cases just to show that the end to end works